code/game/objects/structures.dm

```diff
@@ -97,3 +97,93 @@
 				user << "<span class='warning'>You fail to climb onto [src].</span>"
 			density = 1
 	structureclimber = null
+
+/obj/structure/destructible //a base for destructible structures
+	var/max_health = 100
+	var/health = 100
+	var/takes_damage = TRUE //If the structure can be damaged
+	var/break_message = "<span class='warning'>The strange, admin-y structure breaks!</span>" //The message shown when a structure breaks
+	var/break_sound = 'sound/magic/clockwork/invoke_general.ogg' //The sound played when a structure breaks
+	var/list/debris = list() //Parts left behind when a structure breaks
+
+/obj/structure/destructible/proc/take_damage(amount, damage_type)
+	if(!amount || !damage_type || !damage_type in list(BRUTE, BURN))

```

Comment:
above this line add:
```var/static/list/physical_damage_types = list(BRUTE = TRUE, BURN = TRUE)```
then change: ```|| !damage_type in list(BRUTE,BURN))``` (which is actually wrong too because of ```in```'s precedence), with ```|| !physical_damage_types[damage_type])```